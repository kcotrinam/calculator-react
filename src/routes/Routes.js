import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';
import App from '../components/App';

const Routes = () => (
  <Router>
    <Switch>
      <Route exact path="/" component={App} />
    </Switch>
  </Router>
);

export default Routes;
