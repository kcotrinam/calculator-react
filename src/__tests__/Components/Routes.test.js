import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Routes from '../../routes/Routes';

test('renders correctly', () => {
  const tree = create(<Routes />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('displays heading', () => {
  render(<Routes />);
  const title = screen.getByText('Math Magicians');
  expect(title).toBeInTheDocument();
});

test('Has link to calculator', () => {
  render(<Routes />);
  const a = screen.getByText('Home').closest('a');
  expect(a).toHaveAttribute('href', '/');
});

test('Has link to calculator', () => {
  render(<Routes />);
  const a = screen.getByText('Calculator').closest('a');
  expect(a).toHaveAttribute('href', '/calculator');
});

test('Has link to calculator', () => {
  render(<Routes />);
  const a = screen.getByText('Quote').closest('a');
  expect(a).toHaveAttribute('href', '/quote');
});
