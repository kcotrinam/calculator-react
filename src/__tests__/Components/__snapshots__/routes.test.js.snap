// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="nav"
  >
    <h1
      className="nav-title"
    >
      Math Magicians
    </h1>
    <ul
      className="nav-list"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="home"
  >
    <h1
      className="nav-title"
    >
      Welcome to page!
    </h1>
    <h4
      className="home-quote"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. IpsuLorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
    </h4>
  </div>,
]
`;
